presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -195,10 +193,13 @@ public static boolean castToBoolean(@SqlType(StandardTypes.DOUBLE) double value)
     public static long castToInteger(@SqlType(StandardTypes.DOUBLE) double value)
     {
         try {
-            return toIntExact((long) MathFunctions.round(value));
+            return DoubleMath.roundToInt(value, HALF_UP);
         }
         catch (ArithmeticException e) {
-            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, "Out of range for integer: " + value, e);
+            if (e.getMessage().equals("not in range")) {
```

Comment:
How can this be done without relying on undocumented exception messages that are subject to change between JDKs and JDK versions?

If it can't, I'd drop the conditional and simply always use an INVALID_CAST_ARGUMENT

Reply:
yeah, I didn't love it, but didn't see other options..  I'll switch to always using INVALID_CAST_ARGUMENT instead.